CONTRIBUTING.md

```diff
@@ -0,0 +1,278 @@
+
+When refactoring, structure your PR to make it easy to review and don't hesitate
+to split it into multiple small, focused PRs.
+
+Commits should cover both the issue fixed and the solution's rationale.
+These [guidelines](https://chris.beams.io/posts/git-commit/) should be kept in
+mind.
+
+To facilitate communication with other contributors, the project is making use
+of GitHub's "assignee" field. First check that no one is assigned and then
+comment suggesting that you're working on it. If someone is already assigned,
+don't hesitate to ask if the assigned party or previous commenters are still
+working on it if it has been awhile.
+
+
+## Branches information
+
+The main library development happens in the `master` branch. This branch must
+always compile without errors (using GitHub CI). All external contributions are
+made within PRs into this branch.
+
+Each commitment within a PR to the `master` must
+* compile without errors;
+* contain all necessary tests for the introduced functional;
+* contain all docs.
+
+
+## Peer review
+
+Anyone may participate in peer review which is expressed by comments in the pull
+request. Typically reviewers will review the code for obvious errors, as well as
+test out the patch set and opine on the technical merits of the patch. PR should
+be reviewed first on the conceptual level before focusing on code style or
+grammar fixes.
+
+### Repository maintainers
+
+For the pull request to be merged we require (a) that all CI test should pass 
+and (2) at least two "accepts"/ACKs from the repository maintainers – and no 
+main reasonable "rejects"/NACKs from anybody who reviewed the code.
+
+Current list of the project maintainers:
+
+- [Andrew Poelstra](https://github.com/apoelstra)
+- [Steven Roose](https://github.com/stevenroose)
+- [Maxim Orlovsky](https://github.com/dr-orlovsky)
+- [Matt Corallo](https://github.com/TheBlueMatt)
+- [Elichai Turkel](https://github.com/elichai)
+- [Sebastian Geisler](https://github.com/sgeisler)
+- [Sanket Kanjalkar](https://github.com/sanket1729)
+
+Anyone may become a prepository maintaner by:
+- Demonstrating long-term commitment to the project development (multiple 
+  accepted PRs and code review of other's PRs over prolonged period of time, 
+  proven by the rank in 
+  [GitHub project contributors list](https://github.com/rust-bitcoin/rust-bitcoin/graphs/contributors))
+- Contributing code and reviews at high quality, passing peer-review of the
+  existing maintainers and other participants.
+- Willing to perform peer review and participate in closing issues on the 
+  regular basis in the future.
+
+On the other hand, those, who had stopped participating in the project 
+development on the ongoing basis may be, after some "grace period", be removed
+from the project maintainers, if there are other candidates who can replace them
+(such that "2 positive reviews from maintainers" requirement will remain 
+achievable).
+
+
+## Coding conventions
```

Comment:
Perhaps mention that Rust-idiomatic style is preferred to "how it looks in Core" if everyone agrees.

Reply:
Done below